signer/src/eth.rs

```diff
@@ -98,15 +97,45 @@ impl Keypair {
     }
 }
 
-#[derive(Debug, Clone, Copy, PartialEq, Eq)]
-pub enum Derivation {
-    Hard,
-    Soft,
+/// A derivation path.
+#[derive(Clone, Debug)]
+pub struct DerivationPath {
+    inner: bip32::DerivationPath,
 }
 
-impl Default for Derivation {
-    fn default() -> Self {
-        Self::Soft
+impl DerivationPath {
```

Comment:
Move the derivation path stuff to a separate struct so that we can accept arbitrary derivation strings and have a couple of nice helpers for "common" things